CallAutomation_AppointmentReminder/CallAutomation_AppointmentReminder/Program.cs

```diff
@@ -0,0 +1,136 @@
+using Azure.Communication;
+using Azure.Communication.CallAutomation;
+using Azure.Messaging;
+using CallAutomation_AppointmentReminder;
+using Microsoft.Extensions.FileProviders;
+using static CallAutomation_AppointmentReminder.Logger;
+
+var builder = WebApplication.CreateBuilder(args);
+builder.Services.AddControllers();
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+
+var sourceIdentity = await Identity.CreateUser(builder.Configuration["ConnectionString"]);
+var phoneNumberToAddToCall = builder.Configuration["TargetPhoneNumber"];
+builder.Services.AddSingleton(new CallAutomationClient(builder.Configuration["ConnectionString"]));
+builder.Services.AddSingleton(new CallConfiguration(builder.Configuration["ConnectionString"],
+    sourceIdentity.Id,
+    builder.Configuration["ApplicationPhoneNumber"],
+    builder.Configuration["AppBaseUri"]));
+
+var app = builder.Build();
+
+app.MapPost("/api/call", async (CallAutomationClient callAutomationClient, CallConfiguration callConfiguration) =>
+{
+    var source = new CallSource(new CommunicationUserIdentifier(callConfiguration.SourceIdentity))
+    {
+        CallerId = new PhoneNumberIdentifier(callConfiguration.SourcePhoneNumber)
+    };
+    var target = new PhoneNumberIdentifier(phoneNumberToAddToCall);
+
+    var createCallOption = new CreateCallOptions(source,
+        new List<CommunicationIdentifier>() { target },
+        new Uri(callConfiguration.CallbackEventUri));
+
+    var response = await callAutomationClient.CreateCallAsync(createCallOption).ConfigureAwait(false);
+
+    Logger.LogMessage(MessageType.INFORMATION, $"Reponse from create call: {response.GetRawResponse()}" +
+        $"CallConnection Id : {response.Value.CallConnection.CallConnectionId}");
+});
+
+app.MapPost("/api/callbacks", async (CloudEvent[] cloudEvents, CallAutomationClient callAutomationClient, CallConfiguration callConfiguration) =>
+{
+    // handle callbacks
+    foreach (var cloudEvent in cloudEvents)
+    {
+        CallAutomationEventBase @event = CallAutomationEventParser.Parse(cloudEvent);
+        if (@event is CallConnected)
+        {
+            Logger.LogMessage(MessageType.INFORMATION, $"CallConnected event recieved for callconnetion id: {@event.CallConnectionId}");
+
+            var recognizeOptions =
+            new CallMediaRecognizeDtmfOptions(CommunicationIdentifier.FromRawId(phoneNumberToAddToCall), maxTonesToCollect: 1)
+            {
+                InterruptPrompt = true,
+                InterToneTimeout = TimeSpan.FromSeconds(10),
+                InitialSilenceTimeout = TimeSpan.FromSeconds(5),
+                Prompt = new FileSource(new Uri(callConfiguration.AppBaseUri + Constants.AppointmentReminderMenuAudio)),
+                OperationContext = "AppointmentReminderMenu"
+            };
+            await callAutomationClient.GetCallConnection(@event.CallConnectionId)
```

Comment:
The repetitive nature of GetCallConnection GetCallMedia could be moved after line 46 I think. We can store the CallConnection and CallMedia objects as variables through one 'get' each, then reuse them throughout the conditional logic.